### ISSUE TITLE
Thread_create_sem returns always -1 on Windows

### ISSUE THREAD

**Issue by tpcz** [eclipse-paho/paho.mqtt.c#1064]
**Describe the bug**
For windows target, function Thread_create_sem returns (via out parameter rc) always -1;

**To Reproduce**
Run current code from master branch on Windows

**Expected behavior**
MQTT_Client_create will return -1

**Screenshots**

**Log files**
Please try to attach log files rather than pasting the log contents. It makes the issues easier to read.

** Environment (please complete the following information):**
Windows, current master

**Additional context**

Solution: add 
```
		*rc = (sem == NULL) ? -1 : 0;
```
after CreateEvent call in Thread.c file.

I created pull request, but I do not have signed ECA and I do not want to sing in for now.

**Comment by icraggs** [comment]
Have you seen the call to CreateEvent() fail?

**Comment by tpcz** [comment]
No, but that is not the point.

The point of this bug is that the integer pointed by *rc parameter of `Thread_create_sem` function  is ALWAYS set to -1 when function Thread_create_sem(int *rc) returns on Windows -> MQTTClient_create is returning -1 (i.e, failure) ALWAYS if NO_PERSISTENCE is defined. I.e, return code indicates failure even if everything is fine.


Details:
============
Current implementation (starting Thread.c:190) is as follows:

```
sem_type Thread_create_sem(int *rc)
{
	sem_type sem = NULL;

	FUNC_ENTRY;
	*rc = -1; // Always set return code to -1
        ...     // never touch *rc to the end of the function (on Windows)
}
```

This cause that MQTTClient_createWithOptions() always set rc variable to -1 in MQTTClient.c:490 and following lines and because we are compiling with NO_PERSISTENCE defined, this rc variable is never touched again in MQTTClient_createWithOptions again. Thus, MQTTClient_createWithOptions returns -1 and therefore -> MQTTClient_create() returns -1 -> indicating a failure even though everything went fine.


In summary, on Windows, when compiled witn NO_PERSISTENCE defined, MQTTClient_create returns always `-1` even if everything is fine.

---- 
I quick-fixed this for me by adding `*rc = (sem == NULL) ? -1 : 0` to `Thread_create_sem` after line 202; complete solution could possibly include also check of `rc` after calls to `Thread_create_sem(rc)` in function MQTTClient_createWithOptions (on lines 490 - 493).


Let me know, if you agree that this is something, that should be fixed - I can sign ECA and create pull request with fix.

**Comment by icraggs** [comment]
Oh yes, it should be fixed.  I asked the question about whether you'd seen it fail because it makes a difference in priority if it is affecting practical use of the library.

**Comment by icraggs** [comment]
Fix added.  I preferred to call GetLastError as is done n Thread_create_mutex

**Comment by tpcz** [comment]
Thanks Ian - nevertheless, I believe this helps to set correct error code (in case of error), but the function still returns -1 on windows even if semaphore was created correctly... Maybe Im missing something, but I think line 203 should be: `*rc = (sem == NULL) ? GetLastError() : 0;` - to set rc to 0, if all went OK.
